src/platforms/linux/xdgportal.cpp

```diff
@@ -210,16 +203,24 @@ QString XdgPortal::parseCgroupAppId(const QString& cgroup) {
     QString();
   }
   QString suffix = cgName.sliced(dot+1);
-  QStringList cgSplit = cgName.chopped(dot).split("-");
 
   QString appId;
+  QStringList cgSplit = cgName.first(dot).split("-");
   if (suffix == "service") {
-    // Parse the Application ID if launched via a systemd service.
+    // Systemd services can take the forms:
+    //   app[-<launcher>]-<ApplicationID>-autostart.service (deprecated)
+    //   app[-<launcher>]-<ApplicationID>[@<RANDOM>].service
     if (cgSplit.last() == "autostart") {
       cgSplit.removeLast();
     }
     appId = cgSplit.last().section('@', 0, 0);
   } else if ((suffix == "scope") || (suffix == "slice")) {
+    // Systemd scopes and slices can take the forms:
+    //   app[-<launcher>]-<ApplicationID>-<RANDOM>.scope
+    //   app[-<launcher>]-<ApplicationID>-<RANDOM>.slice
+    if (cgSplit.length() < 3) {
+      return QString();
```

Comment:
Should we log something in all these cases that we're returning a blank string? In the future, it would allow us to better diagnose what is happening from user logs.

Reply:
I don't think every empty string in here needs to be logged, as this method can be called rather frequently (eg: every cgroup creation will trigger it, whether it was created for an application or not). But for scopes, slices and services starting with `app-` it would be safe to log anything that appears malformed.